src/kademlia/dht_storage.cpp

```diff
@@ -346,6 +345,7 @@ namespace
 		}
 
 		void put_immutable_item(sha1_hash const& target
+			, sha1_hash const& our_id
```

Comment:
1. It is unlikely but possible, so the code shouldn't break if it happens.
2. I don't see any reason why mutable items should omit the distance metric. I think this is a bug in the code.